'query' test fails

Here's the alcotest output (using branch https://github.com/rgrinberg/ocaml-semver/pull/3):
![image](https://github.com/rgrinberg/ocaml-semver/assets/343265/6ae60b74-5536-4db4-ad9b-71a5e269fe36)

```
$ make test
dune runtest
File "test/dune", line 2, characters 7-18:
2 |  (name test_semver)
           ^^^^^^^^^^^
Testing `Semver'.
This run has ID `HQR1HU23'.

  [OK]          Semver          0   3 parts.
  [OK]          Semver          1   long parts.
  [OK]          Semver          2   inc.
  [OK]          Semver          3   cmp.
> [FAIL]        Semver          4   query.

┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        Semver          4   query.                                     │
└──────────────────────────────────────────────────────────────────────────────┘
input: 1.1.1; expected result: Some (1.1.1); actual result: None
ASSERT equal
File "test/test_semver.ml", line 65, character 4:
FAIL equal

   Expected: `-1'
   Received: `0'

Raised at Alcotest_engine__Test.check in file "src/alcotest-engine/test.ml", line 200, characters 4-261
Called from Dune__exe__Test_semver.suite.(fun) in file "test/test_semver.ml", line 65, characters 4-41
Called from Alcotest_engine__Core.Make.protect_test.(fun) in file "src/alcotest-engine/core.ml", line 181, characters 17-23
Called from Alcotest_engine__Monad.Identity.catch in file "src/alcotest-engine/monad.ml", line 24, characters 31-35

Logs saved to `/mnt/sandisk256-main/martin/shared/dev/friends/ocaml-semver/_build/default/test/_build/_tests/Semver/Semver.004.output'.
 ──────────────────────────────────────────────────────────────────────────────

Full test results in `/mnt/sandisk256-main/martin/shared/dev/friends/ocaml-semver/_build/default/test/_build/_tests/Semver'.
1 failure! in 0.002s. 5 tests run.
make: *** [Makefile:7: test] Error 1
```

--- a/lib/semver.ml
+++ b/lib/semver.ml
@@ -8,14 +8,16 @@ type version_part = [
   | `Patch
 ]
 
-let compare (l1, l2, l3) (r1, r2, r3) =
+let semver_compare (l1, l2, l3) (r1, r2, r3) =
   match compare l1 r1, compare l2 r2, compare l3 r3 with
   | 0, 0, res -> res
   | 0, res, _ -> res
   | res, _, _ -> res
 
+let compare = semver_compare
+
 let equal a b =
-  compare a b = 0
+  semver_compare a b = 0
 
 let succ p v = match p, v with
   | `Major, (l1, _, _) -> (l1 + 1, 0, 0)
@@ -63,7 +65,9 @@ let query query versions =
     | `Patch (maj, min, patch), v' -> (maj, min, patch) = v'
     | `Minor (maj, min), (v1, v2, _) -> maj = v1 && min = v2
     | `Major maj, (v1, _, _) -> maj = v1 in
-  let res = List.sort compare (List.filter (compare_query query) versions) in
+  let res =
+    List.filter (compare_query query) versions
+    |> List.sort semver_compare in
   if res = []
   then None
   else Some (last res)

--- a/lib/semver.ml
+++ b/lib/semver.ml
@@ -59,12 +59,12 @@ end
 (* Should this just expect a sorted list instead of sorting it itself? *)
 let query query versions =
   let last ls = List.nth ls (List.length ls - 1) in
-  let compareQuery q v = match q, v with
-    | `Patch (maj, min, patch), v' -> (maj, min, patch) == v'
-    | `Minor (maj, min), (v1, v2, _) -> maj == v1 && min == v2
-    | `Major maj, (v1, _, _) -> maj == v1 in
-  let res = List.sort compare (List.filter (compareQuery query) versions) in
-  if res == []
+  let compare_query q v = match q, v with
+    | `Patch (maj, min, patch), v' -> (maj, min, patch) = v'
+    | `Minor (maj, min), (v1, v2, _) -> maj = v1 && min = v2
+    | `Major maj, (v1, _, _) -> maj = v1 in
+  let res = List.sort compare (List.filter (compare_query query) versions) in
+  if res = []
   then None
   else Some (last res)
 

